localization/ekf_localizer/src/ekf_localizer.cpp

```diff
@@ -343,7 +347,11 @@ void EKFLocalizer::callbackPoseWithCovariance(
   if (!is_activated_) {
     return;
   }
-
+  /* Considering change of z value due to NDT delay*/
+  const rclcpp::Time t_curr = this->now();
+  const double delay_time = (t_curr - msg->header.stamp).seconds();
+  const double dz_delay = considering_z_ndt_delay(current_ekf_twist_, delay_time);
+  msg->pose.pose.position.z += dz_delay;
```

Comment:
Please change the implementation here. I'd prefer not to update msg.z with compensated z, since it is kind of a "lie" information.

Reply:
OK.
I will try to implement it in one of the following two functions.
- `measurementUpdatePose()`
- `updateSimple1DFilters()`